packages/hub/routes/notification-preferences.ts

```diff
@@ -0,0 +1,141 @@
+import Koa from 'koa';
+import autoBind from 'auto-bind';
+import { inject } from '@cardstack/di';
+import { ensureLoggedIn } from './utils/auth';
+import NotificationTypeQueries from '../services/queries/notification-type';
+import NotificationPreferenceQueries from '../services/queries/notification-preference';
+import NotificationPreferenceSerializer from '../services/serializers/notification-preference-serializer';
+import { serializeErrors } from './utils/error';
+
+export interface NotificationType {
+  id: string;
+  notificationType: string;
+  defaultStatus: 'enabled' | 'disabled';
+}
+
+export interface NotificationPreference {
+  ownerAddress: string;
+  pushClientId: string;
+  notificationType: string;
+  status: 'enabled' | 'disabled';
+}
+
+export default class NotificationPreferencesRoute {
+  authenticationUtils = inject('authentication-utils', { as: 'authenticationUtils' });
+  notificationTypeQueries: NotificationTypeQueries = inject('notification-type-queries', {
+    as: 'notificationTypeQueries',
+  });
+  notificationPreferenceQueries: NotificationPreferenceQueries = inject('notification-preference-queries', {
+    as: 'notificationPreferenceQueries',
+  });
+  notificationPreferenceSerializer: NotificationPreferenceSerializer = inject('notification-preference-serializer', {
+    as: 'notificationPreferenceSerializer',
+  });
+
+  constructor() {
+    autoBind(this);
+  }
+
+  async get(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let notificationTypes = await this.notificationTypeQueries.query();
+    let notificationPreferences = await this.notificationPreferenceQueries.query({
+      ownerAddress: ctx.state.userAddress,
+      pushClientId: ctx.params.push_client_id,
+    });
+
+    let defaultPreferences = notificationTypes.map((nt) => {
+      return {
+        status: nt.defaultStatus,
+        notificationType: nt.notificationType,
+        ownerAddress: ctx.state.userAddress,
+        pushClientId: ctx.params.push_client_id,
+      };
+    });
+
+    let preferences: NotificationPreference[] = defaultPreferences.map((defaultPreference) => {
+      let preference = notificationPreferences.find(
+        (preference) => preference.notificationType === defaultPreference.notificationType
+      );
+      if (preference) {
+        return preference;
+      } else {
+        return defaultPreference;
+      }
+    });
```

Comment:
This merging/fallback logic should probably be pushed into a model layer.

Reply:
Agree. I'm planning on doing that in a separate PR.